src/utils/httpRequestParser.ts

```diff
@@ -100,7 +114,15 @@ export class HttpRequestParser implements IRequestParser {
         // parse body
         const contentTypeHeader = getHeader(headers, 'content-type') || getHeader(this._restClientSettings.defaultHeaders, 'content-type');
         body = HttpRequestParser.parseRequestBody(bodyLines, requestAbsoluteFilePath, contentTypeHeader);
-        if (this._restClientSettings.formParamEncodingStrategy !== FormParamEncodingStrategy.Never && body && typeof body === 'string' && MimeUtility.isFormUrlEncoded(contentTypeHeader)) {
+        if (isGraphQlRequest) {
+            variables = HttpRequestParser.parseRequestBody(variableLines, requestAbsoluteFilePath, contentTypeHeader);
+
+            let graphQlPayload = {
+                query: body,
+                variables: JSON.parse(variables.toString())

```

Comment:
If no variables provided, do we still need to serialize this, or if we included `variables: null` in the payload, will the GraphQL server allow this?

Reply:
yes, it still needs to be sent. I've never personally used a request without variables, but I'll add a default state